packages/ra-core/src/controller/create/useCreateController.ts

```diff
@@ -94,7 +98,7 @@ export const useCreateController = <
                 onError: onErrorFromSave,
                 transform: transformFromSave,
                 meta: metaFromSave,
-            } = {}
+            } = {} as SaveHandlerCallbacks
```

Comment:
Handled in the dataProvider hook. Moved the side effects at declaration time

Reply:
Just for my understanding, the side effects were previously set at call time, now they're set at definition time. Is there a reason for this change?

Reply:
To avoid side effects not being executed on optimistic modes. Indeed, if the component is unmounted, call time side effects are not executed, only declaration time are. This was the case in react-query v3 and I don't understand how it could work before.